stackslib/src/main.rs

```diff
@@ -974,6 +975,61 @@ simulating a miner.
         process::exit(1);
     }
 
+    if argv[1] == "get-tenure-inv" {
+        let chainstate_root_path = &argv[2];
+        let tip_block_ids = &argv[3..];
+        let chainstate_path = format!("{}/chainstate", &chainstate_root_path);
+        let sortition_path = format!("{}/burnchain/sortition", &chainstate_root_path);
+
+        let (chainstate, _) =
+            StacksChainState::open(false, 0x80000000, &chainstate_path, None).unwrap();
+        let pox_consts =
+            PoxConstants::new(900, 100, 80, 0, 0, u64::MAX, u64::MAX, 240, 241, 242, 242);
+        let sortition_db = SortitionDB::open(&sortition_path, true, pox_consts).unwrap();
+
+        let mut invgen = InvGenerator::new();
+        let tip = SortitionDB::get_canonical_burn_chain_tip(sortition_db.conn()).unwrap();
+
+        for tip_block_id in tip_block_ids.iter() {
+            let tip_block_id = StacksBlockId::from_hex(tip_block_id).unwrap();
+            let header =
+                NakamotoChainState::get_block_header_nakamoto(chainstate.db(), &tip_block_id)
+                    .unwrap()
+                    .unwrap();
+            let sn = SortitionDB::get_block_snapshot_consensus(
+                sortition_db.conn(),
+                &header.consensus_hash,
+            )
+            .unwrap()
+            .unwrap();
+
+            let reward_cycle = sortition_db
+                .pox_constants
+                .block_height_to_reward_cycle(230, sn.block_height)
```

Comment:
Should 230 be put into a variable?

Reply:
This is a one-off tool that I wrote to test this against the live nakamoto testnet.  I'm fine with just dropping the patch to this file.